app/src/main/java/com/nervousfish/nervousfish/service_locator/IServiceLocatorCreator.java

```diff
@@ -0,0 +1,9 @@
+package com.nervousfish.nervousfish.service_locator;
+
+
+/**
+ * Defines the interface of a class that can create a new {@link IServiceLocator}.
+ */
+public interface IServiceLocatorCreator {
```

Comment:
Because every service locator needs its own "creator", otherwise the serviceLocator variable cannot be final